WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/details/ProductDetailViewModel.kt

```diff
@@ -2603,10 +2672,14 @@ class ProductDetailViewModel @Inject constructor(
         val isLoadingMore: Boolean? = null,
         val canLoadMore: Boolean? = null,
         val isRefreshing: Boolean? = null,
-        val isEmptyViewVisible: Boolean? = null
+        val isEmptyViewVisible: Boolean? = null,
+        val searchQuery: String? = null
     ) : Parcelable {
         val isAddCategoryButtonVisible: Boolean
             get() = isSkeletonShown == false
```

Comment:
With the new search logic, we altered how `isSkeletonShown` is updated. 
After several tests I think that makes most sense to me and that works well is adding the following small change
```suggestion
            get() = isRefreshing == false
```

Let me know what you think :)

Reply:
Nice solution! I was curious and my patch was to update `isSkeletonShown` value inside `observeProductCategorySearchQuery()`, but yours is more elegant. Additionally, that also hides the button during loading. Previously it will still be visible and tappable during loading, which I thought was kind of an odd UX.